20987 initial CSS container queries updates (#22186)
mdn/content: files/en-us/web/css/container/index.md

---

> title: Container shorthand

Reviewer: ```suggestion
title: container
```

slug: Web/CSS/container

> page-type: css-property

Reviewer: ```suggestion
page-type: css-shorthand-property
```

tags:
  - container
  - CSS
  - CSS Containment
  - Reference

>   - Property

Reviewer: Not sure if we're still using this:

```
  - recipe:css-shorthand-property
```

Author: noted this one [as above](https://github.com/mdn/content/pull/22186/files#r1018673764)

browser-compat: css.properties.container
---

{{CSSRef}}

> The **container** [CSS](/en-US/docs/Web/CSS) property is shorthand usage for giving a name to a [containment context](/en-US/docs/Web/CSS/CSS_Container_Queries#naming_containment_contexts) used in a [container query](/en-US/docs/Web/CSS/CSS_Container_Queries).

Reviewer: ```suggestion
The **container** [shorthand](/en-US/docs/Web/CSS/Shorthand_properties) [CSS](/en-US/docs/Web/CSS) property establishes the element as a query container and specifies the name or name for the [containment context](/en-US/docs/Web/CSS/CSS_Container_Queries#naming_containment_contexts) used in a [container query](/en-US/docs/Web/CSS/CSS_Container_Queries).
```

## Syntax

```css

> container: <container-name> <container-type>;

Reviewer: ```suggestion
container: <container-name> / <container-type>;
```

Reviewer: This syntax looks like syntax. The previous properties don't have this.

```

## Example

Given the following HTML example which is a card component with an image, a title, and some text:

```html
<div class="container">
  <div class="card">
    <img src="image.png" alt="An awesome picture of a cat" />
    <h2>Card title</h2>
    <p>Card content</p>
  </div>
</div>
```

The more explicit way to create a container context using `container-type` and `container-name` is:

```css
.container {
  container-type: inline-size;
  container-name: sidebar;
}
```

The shorthand for this is the following:

```css
.container {

>   container: sidebar inline-size;

Reviewer: ```suggestion
  container: sidebar / inline-size;
```
